### DISCUSSION TITLE
ZGC support

### DISCUSSION THREAD

**Discussion by vans239** [apangin/nalim#9] [General]
It seems that ZGC doesn't support EnableJVMCI currently and will not support in nearest future

Env: 
- jave 20.0.2
- `-XX:+UseZGC `
```
Setting EnableJVMCI to false as selected GC does not support JVMCI: z gc
```

It also seems like the support was added for non-generational zgc in java 21
Java 20: `bool JVMCIGlobals::gc_supports_jvmci() {  return UseSerialGC || UseParallelGC || UseG1GC;  } `
Java 21: `bool JVMCIGlobals::gc_supports_jvmci() { return UseSerialGC || UseParallelGC || UseG1GC || (UseZGC && !ZGenerational); }`

Are you aware of any workarounds?
Would it be worth to outline in docs that some gcs don't EnableJVMCI?

**Comment by apangin** [comment]
> will not support in nearest future
> ...
> was added for non-generational zgc in java 21

JDK 21 is about a month ahead from today - pretty much nearest future :)
I have no plans to make nalim work in configurations where JVMCI is not supported.

> Would it be worth to outline in docs that some gcs don't EnableJVMCI?

Added a note in README.
